standards/managing-services-interactions.md

```diff
@@ -0,0 +1,45 @@
+---
+category: Building software
+expires: 2018-03-29
+---
+# Managing interactions between services
+
+Services need to interact with one another over APIs. They need to be
+able to find one another, authenticate clients, be fault-tolerant,
+scale appropriately in line with demand, and more.
+
+## Principles
+
+Services should be built on the assumption that [whatever network they
+reside in is
+insecure](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing).
+It should be possible to deploy a new instance of a service to a new
+environment (with appropriate configuration) and have it present an
+appropriately-secured interface to the network.
+
+A failure (including high latency) in a downstream service should not
+inherently mean a failure in a consuming service, but it is not possible
+for a third party service to accurately decide the appropriate response
+in the case of a failure. Only the provider or consumer of the API can
+have enough knowledge of failure states to handle them.
+
+Handling increased load is best done at the level of the service.
+Inserting a caching proxy, rate limiting layer, or other middleware, in
+front of a suite of services assumes that all of those services have
+similar usage and caching characteristics, and that they will need to
+scale in the same fashion. If any one of them diverges from this, the
+middleware will then have to scale in two (probably competing)
+directions.
```

Comment:
This para is a bit opaque to me - is it saying that we shouldn't introduce a single common caching/rate limiting/etc proxy across multiple services?

Reply:
That's what it's trying to say, yeah. I'll see if I can make that a bit clearer.